doc/source/user/numpy-for-matlab-users.rst

```diff
@@ -25,160 +25,46 @@ Some Key Differences
 
 .. list-table::
 
-   * - In MATLAB®, the basic data type is a multidimensional array of
-       double precision floating point numbers.  Most expressions take such
-       arrays and return such arrays.  Operations on the 2-D instances of
-       these arrays are designed to act more or less like matrix operations
-       in linear algebra.
-     - In NumPy the basic type is a multidimensional ``array``.  Operations
-       on these arrays in all dimensionalities including 2D are element-wise
-       operations.  One needs to use specific functions for linear algebra
-       (though for matrix multiplication, one can use the ``@`` operator
-       in python 3.5 and above).
+   * - In MATLAB®, the basic type is a multidimensional array. By default, MATLAB®
+       creates two dimensional arrays of double precision floating point numbers.
+       Operations on the 2D instances of these arrays are designed to act like
+       matrix operations in linear algebra. Most expressions take such arrays and
+       return arrays.
+     - In NumPy, the basic type is a multidimensional ``array``. By default,
+       NumPy creates 0-D or 1-D arrays for inputs of numbers or lists,
```

Comment:
Perhaps confusing, as `np.array([[1]])` is not 1D, even though it is a list.

Reply:
`np.array([[1]]).shape` is `(1,1)` so it is a 2D array. If you assign it to the variable `a=np.array([[1]])`, you can reference the `1` with `a[0,0]`.

This is a tripping point for Matlab users migrating to NumPy. Because in Matlab you would get the number 1 by calling 

`a`, `a(1)`, or `a(1,1)`, 

but in NumPy if you're trying to call the number `1`, you need to call out the value in the multidimensional array

`a[0,0]` -> the number `1`

`a[0]` -> a 1D array with one value, `1`

a -> a 2D array with one value, `1`

Maybe we need a little more details in this description.

Reply:
@eric-wieser, I've got 88 lines of rst (see below) that I could add to walk a matlab user through this idea of multidimensional arrays in numpy. I think it would fit in the Introduction at line 16. I pasted the code in a gist to see what others think. 

[MATLAB vs NumPy array creation defaults](https://gist.github.com/cooperrc/e023ee0cdf125b482a131fa992f329a7)

Reply:
As a quick fix for this line then, perhaps:
```suggestion
       NumPy creates 0-D or 1-D arrays for inputs of numbers or lists of numbers,
```

Reply:
looks good to me, I'll make the change